### PR TITLE
sanitize_267_download_refactor

### LINKED ISSUE #267: Sanitize user input when executing local commands
User input should be sanitized before local command execution, otherwise there is a possibility for command injection.

E.g. see this code
https://github.com/gardener/gardenctl/blob/3ff05d6244a4cc19d618a55037dfd0ac1b9b31b8/pkg/cmd/ssh_alicloud.go#L202

a.MyPublicIP is determined by calling an external service 
https://github.com/gardener/gardenctl/blob/3ff05d6244a4cc19d618a55037dfd0ac1b9b31b8/pkg/cmd/miscellaneous.go#L327-L331

If this service returns a command instead of an IP, I assume that this command would get executed as per code above.
There are a lot of other places where local commands are executed and most of them, if not all, do not sanitize the input. 

And taking another look at the above example, why is the aliyun binary not called directly with the other parameters as arguments? https://github.com/gardener/gardenctl/blob/3ff05d6244a4cc19d618a55037dfd0ac1b9b31b8/pkg/cmd/ssh_alicloud.go#L202

See also https://github.com/gardener/gardenctl/issues/266 to further reduce the need to execute local commands and thereby reducing the attack vector.

### PR DESCRIPTION
**Pull request by tedteng**
**What this PR does / why we need it**:
sanitize bash -c from download 
**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardenctl/issues/267

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: pkg/cmd/download.go [hunk 1]
```diff
@@ -204,8 +204,8 @@ func downloadLogs(option string, targetReader TargetReader) {
 		pathLogsSeeds := filepath.Join(dir, "seeds", *shoot.Spec.SeedName, shoot.ObjectMeta.GetNamespace(), shoot.Name, "logs", "vpn")
 		for _, pod := range pods.Items {
 			if strings.Contains(pod.Name, "prometheus-0") {
-				fmt.Println("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
-				output, err := ExecCmdReturnOutput("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
+				fmt.Println("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
+				output, err := ExecCmdReturnOutput("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
```
Comment (petersutter) [inline pkg/cmd/download.go:208]: This does not work. Please properly pass the arguments to the function

File: pkg/cmd/download.go [hunk 2]
```diff
@@ -217,8 +217,8 @@ func downloadLogs(option string, targetReader TargetReader) {
 				}
 			}
 			if strings.Contains(pod.Name, "kube-apiserver") {
-				fmt.Println("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
-				output, err := ExecCmdReturnOutput("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
+				fmt.Println("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
+				output, err := ExecCmdReturnOutput("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -c "+"vpn-seed"+" -n "+shoot.Status.TechnicalID)
```
Comment (petersutter) [inline pkg/cmd/download.go:221]: Same here, this does not work

File: pkg/cmd/download.go [hunk 3]
```diff
@@ -260,8 +260,8 @@ func downloadLogs(option string, targetReader TargetReader) {
 		pathLogsShoots := filepath.Join(dir, "seeds", *shoot.Spec.SeedName, shoot.ObjectMeta.GetNamespace(), shoot.Name, "logs", "vpn")
 		for _, pod := range pods.Items {
 			if strings.Contains(pod.Name, "vpn-shoot-") {
-				fmt.Println("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -n "+"kube-system")
-				output, err := ExecCmdReturnOutput("bash", "-c", "export KUBECONFIG="+KUBECONFIG+"; kubectl logs "+pod.Name+" -n "+"kube-system")
+				fmt.Println("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -n "+"kube-system")
+				output, err := ExecCmdReturnOutput("kubectl", "--kubeconfig="+KUBECONFIG, "logs "+pod.Name+" -n "+"kube-system")
```
Comment (petersutter) [inline pkg/cmd/download.go:264]: Same here, this does not work

**Review by petersutter — CHANGES_REQUESTED** [review]

**Comment by tedteng** [comment]
all fixed

**Review by petersutter — APPROVED** [review]
/lgtm and much better, thank you

**Comment by neo-liang-sap** [comment]
@tedteng please sign the CLA for this PR and all other your opening PRs.
besides please squash commits and i will proceed with merge

**Comment by petersutter** [comment]
there seems to be an issue with the cla-assistant. 
> To trigger the recheck of a PR open `https://cla-assistant.io/check/{organisation}/{repo_name}?pullRequest={number}`. If successful you should be redirected to github. Sometimes it takes a moment but it should change to a green checkmark.

**Pull request merged**